enhancements/authentication/configuring-webhook-token-authenticators.md

```diff
@@ -0,0 +1,176 @@
+for the integrated oauth-server, which might be helpful in cases where we would like to
+delegate authentication to a different OpenShift control plane, or direct authentication to an
+identity provider capable of handling the Kubernetes token reviews, such as Keycloak.
+
+### Goals
+
+1. Make it possible to configure an endpoint capable of performing kube token reviews
+2. Modify cluster-authentication-operator so that it configures authentication via the OpenShift OAuth
+   server by using the `webhookTokenAuthenticator` field instead of the current patching of the
+   kube-apiserver.
+
+### Non-Goals
+
+1. provide a partial authentication operator to any external authenticator.  Once the standard oauth-server
+   is disabled, authentication configuration is *entirely* the responsibility of the new authentication stack.
+2. allow both the built-in oauth-server and another authentication stack to be enabled at the same time.
+
+## Proposal
+
+### Implementation
+
+#### API
+
+`authentications.config.openshift.io/v1` already contains the `spec.webhookTokenAuthenticators`. It
+expects an array of references to secrets, but upstream only allows a single authenticator. This field
+will therefore be deprecated in favour of a new one, `spec.webhookTokenAuthenticator`.
+
+The secret reference which gets set in this field has to be validated. The referenced secret will be
+expected to contain a `kubeconfig` key with a valid kubeconfig configuration for the webhook. In 4.5,
+it will only be possible to configure a single element for the `wehbhookTokenAuthenticator` field
+and the referenced kubeconfigs will only be allowed to use the `-data` forms (e.g.
+`certificate-authority-data` is allowed whereas `certificate-authority` is not) of their fields to
+configure certificate authority, client certificate and key.
+
+This path is compatible with kubernetes as it is possible to directly map the referenced
+secret's contents to files and specify these in the kube-apiserver configuration.
+
+#### Backend - kube-apiserver-operator
+
+1. Observes any changes in the `webhookTokenAuthenticator` field (and changes to the referenced secret)
+   of the `authentication/cluster` resource and performs validation of the reference in that field.
+2. It pushes the webhook authenticator, iff it passes the validation to the `webhook-authenticator`
+   secret in `openshift-kube-apiserver` namespace.
+3. If the webhook authenticator does not pass validation, the operator keeps the previous state.
+4. Redeploys the kube-apiserver static pods while specifying the new
+   mounted file path in the `apiServerArguments.authentication-token-webhook-config-file`
+
+#### Backend - Integrated OAuth Server Token Authentication
+
+1. cluster-authentication-operator observes `authentication.config/cluster` `type` field
+2. if set to "IntegratedOAuth", it creates an `openshift-config/webhook-authentication-integrated-oauth`
```

Comment:
what for other values? It removes itself? Or just leaves everything configured?

Reply:
Let's stick with noop, have the user do their thing.